src/java/org/apache/cassandra/db/commitlog/CommitLogReplayer.java

```diff
@@ -300,11 +310,13 @@ protected void onInvalidMutation(TableId id)
             logger.debug("Invalid mutation detected for table id {}", id);
         }
 
-        protected Future<Integer> initiateMutation(final Mutation mutation,
-                                                   final long segmentId,
-                                                   final int serializedSize,
-                                                   final int entryLocation,
-                                                   final CommitLogReplayer commitLogReplayer)
+        protected void onFailedMutation(String keyspace, Collection<TableId> tableIds) {}
```

Comment:
I'd rather log error in the future exception handling (https://github.com/datastax/cassandra/pull/1228/files/200e64c9cc868acef2cc4668e496d6647d6a5b8e#diff-1f85e431ca2a0820447e3eb25b71be47a37dfe6ff5597d9b725c0cc9448a7987R377) , otherwise we'd need to pass an exception here. Also if we only log here, then subclass has to always call super method.

Reply:
Fair enough.